pkg/sql/expand_plan.go

```diff
@@ -498,8 +498,9 @@ func simplifyOrderings(plan planNode, usefulOrdering sqlbase.ColumnOrdering) pla
 		n.ordering.trim(usefulOrdering)
 
 	case *renderNode:
-		n.ordering.trim(usefulOrdering)
 		n.source.plan = simplifyOrderings(n.source.plan, translateOrdering(usefulOrdering, n))
+		// Recompute r.ordering using the source's simplified ordering.
+		n.ordering = n.computeOrdering(n.source.plan.Ordering())
```

Comment:
This little change isn't really necessary but it seems more natural to do it this way. There is a difference in the case where the "outer" ordering maps to an "inner" ordering that maps back to a different "outer" ordering but it's a non-functional difference - we won't be able to use the ordering either way.

A more general fix would be to also pass the desired/useful ordering to `computeOrdering` so that when it has a choice, it chooses something that matches the given ordering more. In that case, we would want to do it this way, as `trim()` wouldn't have that capability.